src/core/mac/sub_mac.cpp

```diff
@@ -201,6 +202,11 @@ void SubMac::HandleReceiveDone(RxFrame *aFrame, otError aError)
         mPcapCallback(aFrame, false, mPcapCallbackContext);
     }
 
+    if (!ShouldHandleTransmitSecurity() && aFrame != NULL && aFrame->mInfo.mRxInfo.mAckedWithSecEnhAck)
```

Comment:
That's a good question.

In general, how would we handle the enhanced ack for a received frame in the case where OT (`SubMac`) is handling the security frame counter and the counter is not given to the radio platform layer (from `SetFrameCounter()` below). 
Do we not allow any enhanced ack in that case?

Reply:
I think we can think more on this and address it later (after merging this PR).

Reply:
There are two cases:

1. Radio or spinel_radio supports `OPENTHREAD_CONFIG_MAC_SOFTWARE_TX_SECURITY_ENABLE`, then the frame counter is maintained by radio layer. In `SubMac`, this metainfo should be used to update the counter and invoke NVM save if needed.

2. Radio or spinel_radio does not supports `OPENTHREAD_CONFIG_MAC_SOFTWARE_TX_SECURITY_ENABLE`, then the frame counter is maintained by `SubMac`. Enh-Ack is also generated at `SubMac` which should be using and incrementing the frame counter there. `SubMac` should be the source of the truth rather than rely on the information from lower layer.